### PR TITLE
[MPS] torch.nn.functional.pad may cause broken tensor if elements >= 2^16

### PR DESCRIPTION
**Pull request by niw**
**Problem**

It's undocumented limitation, or probably a bug though, based on the test on actual device,
`-[MPSGraph padTensor:withPaddingMode:leftPadding:rightPadding:constantValue:name:]` API doesn't work properly if number of elements >= 2^16 (65536) and returns a broken tensor, which elements are sometimes 0.0, sometimes completely random value.

It is tested with this Swift code on the following environments: https://gist.github.com/niw/456d83732bd2c720359abfab1e6eae0f
- MacBook Pro, M1, macOS 15.1
- MacBook Pro, M3, macOS 14.6.1, macOS 15.0.1

Originally this problem is found at kijai/ComfyUI-MochiWrapper#66.

**Solution**

Check the number of elements in `pad_out_template` and throw not implemented error if it's >= 2^16.

Add a unit test that should success but not on actual device, catch not implemented error and assert it for now.

cc @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen

### CODE REVIEW

**Comment by niw** [comment]
Possibly related to #135442 or #135447

**Comment by hvaara** [comment]
There's been quite a few similar issues in the past. Additional possibly related issues: #129207 #96982 #134177 

These issues may manifest as silent correctness bugs. We should consider adding unit tests for more ops with the edge cases around 2^16 and 2^32 in the dims (eg. `{2^16, 2^16 ± 1}`). Correctness bugs are very frustrating for end-users, because they're difficult to track down as the buggy ops are often part of a larger system. With good ops coverage we would discover more of these bugs before releases which, I'd argue, is far favorable to having to rely on the wider PyTorch community to report them one-by-one as they appear in the wild.

@pytorchbot label 'module: mps' 'module: correctness (silent)' 'ciflow/mps'

**Comment by hvaara** [comment]
@malfet Does this PR LGTY?

**Comment by niw** [comment]
Ooh, time to rebase on top of current tip...!
